include/cuco/detail/__config

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11000) && defined(__CUDA_ARCH__) && \
+  (__CUDA_ARCH__ >= 700)
```

Comment:
This can use the `and` logic of `CUCO_HAS_CG_MEMCPY_ASYNC` and `CUCO_HAS_LARGE_TYPE_SUPPORT`

Reply:
Actually, I'm not sure if we want this macro depending on `CUCO_HAS_CG_MEMCPY_ASYNC` and `CUCO_HAS_LARGE_TYPE_SUPPORT`.

Reply:
Yes, would be possible but I don't see the semantic relation between these flags.

Reply:
Agreed.